app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt

```diff
@@ -0,0 +1,117 @@
+package uk.gov.android.authentication
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.ActivityCompat
+import java.util.UUID
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationRequest
+import net.openid.appauth.AuthorizationResponse
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.AuthorizationServiceConfiguration
+
+@Suppress("TooGenericExceptionThrown")
+class AppAuthSession : LoginSession {
+    private var context: Context? = null
+    private lateinit var authService: AuthorizationService
+
+    override fun init(
+        context: Context
+    ): LoginSession {
+        if (this.context == null) {
+            this.context = context
+            authService = AuthorizationService(context)
+        }
+        return this
+    }
+
+    override fun present(
+        configuration: LoginSessionConfiguration
+    ) {
+        if (context == null) {
+            throw Error("Context is null, did you call init?")
+        }
+
+        with(configuration) {
+            val context = this@AppAuthSession.context!!
+            val nonce = UUID.randomUUID().toString()
+
+            val serviceConfig = AuthorizationServiceConfiguration(
+                authorizeEndpoint,
+                tokenEndpoint
+            )
+
+            val builder = AuthorizationRequest.Builder(
+                serviceConfig,
+                clientId,
+                responseType,
+                redirectUri
+            ).also {
+                it.apply {
+                    setScopes(scopes)
+                    setUiLocales(locale)
+                    setNonce(nonce)
+                    setAdditionalParameters(
+                        mapOf(
+                            "vtr" to vectorsOfTrust
+                        )
+                    )
+                }
+            }
```

Comment:
Most builder patterns usually return itself with the functions it has. With this in mind, we shouldn't need to use an `also` and `apply` block, because the functions, such as `setScopes` use the `apply` block themselves. Are we able to make the Builder have a fluent API style by returning itself?

Reply:
Agreed, will address seperately